shared/styles/css.d.ts

```diff
@@ -1,516 +1,35 @@
-  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline'
-  alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline'
-  alignContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around'
-  overflow?: 'visible' | 'hidden' | 'scroll' | 'auto'
-  flex?: number
-  flexGrow?: number
-  flexShrink?: number
-  flexBasis?: number | string
-  aspectRatio?: number
-  zIndex?: number
-  direction?: 'inherit' | 'ltr' | 'rtl'
-  transform?: ReadonlyArray<
-    | {perspective: number}
-    | {rotate: string}
-    | {rotateX: string}
-    | {rotateY: string}
-    | {rotateZ: string}
-    | {scale: number}
-    | {scaleX: number}
-    | {scaleY: number}
-    | {translateX: number}
-    | {translateY: number}
-    | {translate: [number, number]}
-    | {skewX: string}
-    | {skewY: string}
-    | {matrix: ReadonlyArray<number>}
-  >
-  shadowColor?: Color
-  shadowOffset?: {
-    width?: number
-    height?: number
-  }
-  shadowOpacity?: number
-  shadowRadius?: number
-  backfaceVisibility?: 'visible' | 'hidden'
-  backgroundColor?: Color
-  borderColor?: Color
-  borderTopColor?: Color
-  borderRightColor?: Color
-  borderBottomColor?: Color
-  borderLeftColor?: Color
-  borderRadius?: number
-  borderTopLeftRadius?: number
-  borderTopRightRadius?: number
-  borderBottomLeftRadius?: number
-  borderBottomRightRadius?: number
-  borderStyle?: 'solid' | 'dotted' | 'dashed'
-  borderWidth?: number
-  borderTopWidth?: number
-  borderRightWidth?: number
-  borderBottomWidth?: number
-  borderLeftWidth?: number
-  opacity?: number
-  elevation?: number
-  color?: Color
-  fontFamily?: string
-  fontSize?: number
-  fontStyle?: 'normal' | 'italic'
-  fontWeight?: 'normal' | 'bold' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900'
-  fontVariant?: ReadonlyArray<
-    'small-caps' | 'oldstyle-nums' | 'lining-nums' | 'tabular-nums' | 'proportional-nums'
-  >
-  textShadowOffset?: {
-    width: number
-    height: number
-  }
-  textShadowRadius?: number
-  textShadowColor?: Color
-  letterSpacing?: number
-  lineHeight?: number
-  textAlign?: 'auto' | 'left' | 'right' | 'center' | 'justify'
-  textAlignVertical?: 'auto' | 'top' | 'bottom' | 'center'
-  includeFontPadding?: boolean
-  textDecorationLine?: 'none' | 'underline' | 'line-through' | 'underline line-through'
-  textDecorationStyle?: 'solid' | 'double' | 'dotted' | 'dashed'
-  textDecorationColor?: Color
-  writingDirection?: 'auto' | 'ltr' | 'rtl'
-  tintColor?: Color
-  overlayColor?: string
+type _StylesCrossPlatform = {
+  [k in keyof _StylesDesktop]: k extends keyof _StylesCrossPlatformOverride // use override
+    ? _StylesCrossPlatformOverride[k] // or if its shared between desktop and mobile choose one which extends the other
+    : k extends keyof _StylesMobile
+    ? _StylesMobile[k] extends _StylesDesktop[k]
+      ? _StylesMobile[k]
```

Comment:
we prefer mobile to desktop as its more constrained. the css styles from react mostly are of the form
` 'left' | 'right' | undefined | string ` that last `| string` makes that type just string which is very annoying